crates/verify/src/bytecode.rs

```diff
@@ -0,0 +1,336 @@
+            .ok_or_eyre("Contract artifact not found locally")?;
+
+        let bytecode = artifact
+            .get_bytecode_object()
+            .ok_or_eyre("Contract artifact does not have bytecode")?;
+
+        let bytecode = match bytecode.as_ref() {
+            BytecodeObject::Bytecode(bytes) => bytes,
+            BytecodeObject::Unlinked(_) => {
+                eyre::bail!("Unlinked bytecode is not supported for verification")
+            }
+        };
+
+        // Cmp creation code with locally built bytecode and maybe_creation_code
+        let res = try_match(maybe_creation_code, bytecode, self.verification_type)?;
+        tracing::info!("Match: {} | Type: {:?}", res.0, res.1);
+
+        // TODO: @Yash
+        // Fork the chain at `simulation_block`, deploy the contract and compare the runtime
+        // bytecode.
+        let simulation_block = match self.block {
+            Some(block) => match block {
+                BlockId::Number(BlockNumberOrTag::Number(block)) => block,
+                _ => {
+                    eyre::bail!("Invalid block number");
+                }
+            },
+            None => {
+                let provider = utils::get_provider(&config)?;
+                let creation_block =
+                    provider.get_transaction(creation_data.transaction_hash.to_ethers()).await?;
+                match creation_block {
+                    Some(tx) => tx.block_number.unwrap().as_u64(),
+                    None => {
+                        eyre::bail!(
+                            "Failed to get block number of the contract creation tx, specify using
+        the --block flag"
+                        );
+                    }
+                }
+            }
+        };
+
+        // Fork the chain at `simulation_block`
+        let fork = CreateFork {
+            enable_caching: false,
+            url: self.rpc_url.unwrap_or_default(),
+            env: Default::default(),
+            evm_opts: EvmOpts { fork_block_number: Some(simulation_block), ..Default::default() },
+        };
+        tracing::info!("Forking the chain at block {}", simulation_block);
+        let multi_fork = MultiFork::spawn();
+        let (fork_id, _shared_backend, _env) = multi_fork.create_fork(fork)?;
+
+        tracing::info!("Created fork with id {}", fork_id);
+        // TODO: @Yash
+        // Deploy the contract on the forked chain
+        let fork_rpc_url = multi_fork.get_fork_url(fork_id)?;
+        let mut fork_config = config.clone();
+        fork_config.eth_rpc_url = fork_rpc_url;
+        let fork_provider = utils::get_provider(&fork_config)?;
+
+        let tx = Eip1559TransactionRequest {
+            from: Some(creation_data.contract_creator.to_ethers()),
+            data: Some(bytecode.to_owned().to_ethers()),
+            ..Default::default()
+        };
+        // @mattsse - Need some help here.
+        let pending_tx = fork_provider.send_transaction(tx, None).await?;
```

Comment:
You are trying to send transaction directly to the fork_rpc_url, you need to spin-up a local `Backend` + `Executor` to simulate execution locally. I think `cast run` code might be helpful here

Reply:
Got it. I'll give it a shot.

Reply:
Honestly don't really see the point of this step. If we already got a match for creation code what's the reason for checking runtime codes by simulating creation transaction on the same block, wouldn't outcome always be the same as the real creation transaction?

Reply:
Yes you're right, I think it can be removed.